web/src/components/storage/ProposalSpacePolicySection.jsx

```diff
@@ -0,0 +1,444 @@
+    const policy = settings.spacePolicy;
+    const devices = settings.installationDevices.map(d => d.name);
+    let currentExpanded = devices.filter(d => expandedDevices.includes(d));
+
+    if (policy === "custom" && !autoExpanded) {
+      currentExpanded = [...devices];
+      setAutoExpanded(true);
+    } else if (policy !== "custom" && autoExpanded) {
+      setAutoExpanded(false);
+    }
+
+    if (currentExpanded.sort().toString() !== expandedDevices.sort().toString()) {
+      setExpandedDevices(currentExpanded);
+    }
+  }, [autoExpanded, expandedDevices, setAutoExpanded, setExpandedDevices, settings]);
+
+  const renderRows = () => {
+    const rows = [];
+
+    settings.installationDevices?.forEach((device, index) => {
+      const isExpanded = expandedDevices.includes(device.name);
+      const children = device.partitionTable?.partitions;
+
+      const onCollapse = () => {
+        const otherExpandedDevices = expandedDevices.filter(name => name !== device.name);
+        const expanded = isExpanded ? otherExpandedDevices : [...otherExpandedDevices, device.name];
+        setExpandedDevices(expanded);
+      };
+
+      rows.push(
+        <DeviceRow
+          key={device.name}
+          device={device}
+          settings={settings}
+          rowIndex={rows.length}
+          setSize={children?.length || 0}
+          posInSet={index}
+          isExpanded={isExpanded}
+          onCollapse={onCollapse}
+          onChange={onChange}
+        />
+      );
+
+      children?.forEach((child, index) => {
+        rows.push(
+          <DeviceRow
+            key={child.name}
+            device={child}
+            settings={settings}
+            rowIndex={rows.length}
+            level={2}
+            posInSet={index}
+            isHidden={!isExpanded}
+            onCollapse={onCollapse}
+            onChange={onChange}
+          />
+        );
+      });
+    });
+
+    return rows;
+  };
+
+  return (
+    <Table isTreeTable variant="compact" aria-label={_("Actions to find space")}>
+      <Thead>
+        <Tr>
+          <Th>{columnNames.device}</Th>
+          <Th>{columnNames.content}</Th>
+          <Th>{columnNames.size}</Th>
+          <Th>{columnNames.details}</Th>
+          <Th>{columnNames.action}</Th>
+        </Tr>
+      </Thead>
+      <Tbody>{renderRows()}</Tbody>
+    </Table>
+  );
+};
+
+/**
+ * Widget to allow user picking desired policy to make space
+ * @component
+ *
+ * @param {object} props
+ * @param {SpacePolicy} props.currentPolicy
+ * @param {(policy: string) => void} [props.onChange]
+ */
+const SpacePolicyPicker = ({ currentPolicy, onChange = noop }) => {
+  return (
+    <OptionsPicker>
+      {SPACE_POLICIES.map((policy) => {
+        return (
+          <OptionsPicker.Option
+            key={policy.id}
+            title={policy.label}
+            body={policy.description}
+            onClick={() => onChange(policy.id)}
+            isSelected={currentPolicy?.id === policy.id}
```

Comment:
I see `?.` here unnecessary

Reply:
I prefer keeping it if there is any chance to have a proposal without a selected policy. Now or in the future.